test/trento/application/projectors/database_projector_test.exs

```diff
@@ -264,4 +265,33 @@ defmodule Trento.DatabaseProjectorTest do
 
     assert deregistered_at == projection.deregistered_at
   end
+
+  test "should remove a database instance from the read model after a deregistration" do
+    deregistered_at = DateTime.utc_now()
+
+    insert(:database, id: sap_system_id = Faker.UUID.v4())
+
+    %{instance_number: instance_number, host_id: host_id} =
+      insert(:database_instance, sap_system_id: sap_system_id)
```

Comment:
I would be a bit more exhaustive here, add more instances, and check that they are not removed.

Reply:
Agreed. I added 4 additional instances of both tests. Hope you don't mind that I'm checking the count instead of comparing the whole content as this seems simpler. I can change it otherwise.